### PR TITLE
EQL: Use the implicit tiebreaker sort value in the search queries

### PR DESCRIPTION
**Pull request by astefan**
Following the recent addition of a [default tiebreaker for PIT requests](https://github.com/elastic/elasticsearch/pull/68833), EQL sequence queries that use `search_after` should also include in the list of its values (a timestamp and an optional EQL tiebreaker) the additional default tiebreaker value.

### CODE REVIEW

**Comment by astefan** [comment]
@elasticmachine update branch

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/Ordinal.java [hunk 1]
```diff
@@ -59,16 +66,22 @@ public int compareTo(Ordinal o) {
...
+                }
             }
             // this tiebreaker is null
             else {
                 // nulls are last so unless both are null (equal)
                 // this ordinal is greater (after) then the other tiebreaker
                 // so fall through to 1
                 if (o.tiebreaker == null) {
-                    return 0;
+                    return Long.valueOf(implicitTiebreaker).compareTo(Long.valueOf(o.implicitTiebreaker));
```
Comment (costin) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/Ordinal.java:84]: `Long.compare(implicitTieBreaker, o.implicitTiebreaker)`

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/extractor/ImplicitTiebreakerHitExtractor.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    public boolean equals(Object obj) {
+        if (obj == null || obj.getClass() != getClass()) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        return 35;
```
Comment (costin) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/extractor/ImplicitTiebreakerHitExtractor.java:58]: ? Ideally should be a prime such as `31` or a relative constant such as `ImplicitTiebreakerHitExtractor.class.hashCode()`

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/Criterion.java [hunk 3]
```diff
@@ -89,7 +91,13 @@ public Ordinal ordinal(SearchHit hit) {
             }
             tbreaker = (Comparable<Object>) tb;
         }
-        return new Ordinal(timestamp, tbreaker);
+
+        Object implicitTbreaker = implicitTiebreaker.extract(hit);
+        if (implicitTbreaker instanceof Number == false) {
+            throw new EqlIllegalArgumentException("Expected _shard_doc/implicit tiebreaker as long but got [{}]", implicitTbreaker);
```
Comment (bpintea) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/Criterion.java:97]: Regarding the error message, could the value be a non-long integer?

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/Ordinal.java [hunk 4]
```diff
@@ -59,16 +66,22 @@ public int compareTo(Ordinal o) {
         }
         if (timestamp == o.timestamp) {
             if (tiebreaker != null) {
-                // if the other tiebreaker is null, it is higher (nulls are last)
-                return o.tiebreaker != null ? tiebreaker.compareTo(o.tiebreaker) : -1;
+                if (o.tiebreaker != null) {
+                    if (tiebreaker.compareTo(o.tiebreaker) == 0) {
+                        return Long.valueOf(implicitTiebreaker).compareTo(Long.valueOf(o.implicitTiebreaker));
```
Comment (bpintea) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/Ordinal.java:71]: I guess Costin's longs comparison applies here as well.

**Review by bpintea — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/Criterion.java (hunk 3 above)
Comment (astefan) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/Criterion.java:97]: @bpintea no. Always `long`.

Comment (bpintea) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/Criterion.java:97]: Would then the instance check agains a `Long` be also appropriate? (just FMI, its prolly slightly safer the way it is now)

Comment (astefan) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/Criterion.java:97]: I'd leave it as is, for the reason you mentioned but also to be consistent with the approach throughout that method.

**Review by matriv — APPROVED** [review]
LGTM. nice!

**Pull request merged**